test/features/search/search_email_filter_test.dart

```diff
@@ -0,0 +1,169 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter_operator.dart';
+import 'package:jmap_dart_client/jmap/core/filter/operator/logic_filter_operator.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/core/utc_date.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_filter_condition.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/mailbox/presentation_mailbox.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/email_receive_time_type.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/email_sort_order_type.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/search_email_filter.dart';
+import 'package:tmail_ui_user/features/thread/domain/model/search_query.dart';
+
+void main() {
+  group('SearchEmailFilter::test', () {
+    group('generateFilterFromToField::test', () {
+      late SearchEmailFilter searchEmailFilter;
+
+      setUp(() {
+        searchEmailFilter = SearchEmailFilter();
+      });
+
+      test('SHOULD generate a single filter WHEN "to" has one item', () {
+        // Arrange
+        searchEmailFilter = SearchEmailFilter(
+          to: {'singleRecipient@example.com'},
+        );
+
+        // Act
+        final filters = searchEmailFilter.generateFilterFromToField();
+
+        // Assert
+        expect(filters.length, 1);
+
+        expect(filters.first, isA<LogicFilterOperator>());
+
+        final filter = filters.first as LogicFilterOperator;
+        expect(filter.operator, Operator.OR);
+
+        expect(filter.conditions, {
+          EmailFilterCondition(to: 'singleRecipient@example.com'),
+          EmailFilterCondition(cc: 'singleRecipient@example.com'),
+          EmailFilterCondition(bcc: 'singleRecipient@example.com'),
+        });
+      });
+
+      test('SHOULD generate multiple filters WHEN "to" has multiple items', () {
+        // Arrange
+        searchEmailFilter = SearchEmailFilter(
+          to: {'recipient1@example.com', 'recipient2@example.com'},
+        );
+
+        // Act
+        final filters = searchEmailFilter.generateFilterFromToField();
```

Comment:
please verify this filter has AND operator

Reply:
It added to `mappingToEmailFilterCondition` unit test